domain/user/state/state.go

```diff
@@ -219,6 +228,51 @@ WHERE user.name = $M.name AND removed = false
 	return usr, nil
 }
 
+// GetUserWithAuth will retrieve the user with checking authentication information
+// specified by UUID from the database. If the user does not exist
+// an error that satisfies usererrors.NotFound will be returned.
+func (st *State) GetUserWithAuth(ctx context.Context, uuid user.UUID) (user.User, error) {
+	db, err := st.DB()
+	if err != nil {
+		return user.User{}, errors.Annotate(err, "getting DB access")
+	}
+
+	var usr user.User
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		getUserWithAuthQuery := `
+SELECT (
+        user.uuid, user.name, user.display_name, user.created_by_uuid, user.created_at, 
+        user_password.password_hash, user_password.password_salt
+       ) AS (&User.*)
+FROM user
+JOIN user_password ON user.uuid = user_password.user_uuid
+WHERE user.uuid = $M.uuid AND removed = false
```

Comment:
`AND user.password = $M.password_hash`

Reply:
I cannot add this condition to the current transaction because to generate `hash`, I need `salt.` So I can do it either in two transactions or checking the hash equality in a manner that I've already implemented in the state function.